OCE-55: Add topology(link) support to the OpenNMS Direct datasource

mattixtech opened the pull request:
- Added support to the integration API for clients to receive edges from OpenNMS topology

Note: I didn't bump the version, not sure if that is required for these changes.

Tested manually during feature test of links over the API.

Jira: https://issues.opennms.org/browse/OCE-55

j-white commented on api/src/main/java/org/opennms/integration/api/v1/topology/TopologyEdgeConsumer.java:
```diff
@@ -0,0 +1,59 @@
+     *
+     * @return the comma separated list of protocols this consumer supports
+     */
+    String getProtocols();
```
Can we make it possible to register for all protocols without having to specify them?

j-white commented on api/src/main/java/org/opennms/integration/api/v1/model/TopologySegment.java:
```diff
@@ -0,0 +1,39 @@
+public interface TopologySegment extends TopologyRef {
+    String getProtocol();
+
+    String getSegmentCriteria();
```
Can we document that the segment criteria is?

j-white commented on api/src/main/java/org/opennms/integration/api/v1/dao/EdgeDao.java:
```diff
@@ -0,0 +1,70 @@
+     * @param protocol the protocol to filter by
+     * @return the count of edges corresponding to the given protocol
+     */
+    long getEdgeCount(String protocol);
```
Can we make the protocol an enum instead of a string?

mattixtech commented:
Addressed the feedback, will be updating the Opennms and OCE code as well to handle the changes. Will re-test everything once all the feedback has been received.

mattixtech commented:
Going to merge this now, and if there are further changes necessary after more testing I will open a new PR.
